op-e2e/eip4844_test.go

```diff
@@ -108,11 +122,37 @@ func TestSystem4844E2E(t *testing.T) {
 		// wait for chain to be marked as "safe" (i.e. confirm batch-submission works)
 		stat, err := rollupClient.SyncStatus(context.Background())
 		require.NoError(t, err)
-		return stat.SafeL2.Number > 0
+		return stat.SafeL2.Number >= receipt.BlockNumber.Uint64()
 	}, time.Second*20, time.Second, "expected L2 to be batch-submitted and labeled as safe")
 
 	// check that the L2 tx is still canonical
 	seqBlock, err = l2Seq.BlockByNumber(context.Background(), receipt.BlockNumber)
 	require.NoError(t, err)
 	require.Equal(t, seqBlock.Hash(), receipt.BlockHash, "receipt block must match canonical block at tx inclusion height")
+
+	// find L1 block that contained the blob(s) batch tx
+	tip, err := l1Client.HeaderByNumber(context.Background(), nil)
+	require.NoError(t, err)
+	var numBlobs int
+	_, err = gethutils.FindBlock(l1Client, int(tip.Number.Int64()), 0, 5*time.Second,
+		func(b *types.Block) (bool, error) {
+			for _, tx := range b.Transactions() {
+				if tx.Type() != types.BlobTxType {
+					continue
+				}
+				numBlobs = len(tx.BlobHashes())
+				// expect to find at least one tx with more than 1 blob in multi-blob case
+				if !multiBlob || numBlobs > 1 {
```

Comment:
is it worth also checking to make sure blobs other than the last one are fully utilized?

Reply:
Definitely something to test! But maybe we better do this in a unit test for the `ChannelBuilder`?

Reply:
either way (or both) works for me.

Reply:
added these assertions to the test